### DISCUSSION TITLE
Use of DataTypeDefinition

### DISCUSSION THREAD

**Discussion by AndreyKl4** [FreeOpcUa/python-opcua#1253] [Q&A] [answered]
Hi all!

I know that in opcua-python is opportunity to define custom data types with help DataTypeDictionary (https://reference.opcfoundation.org/v104/Core/docs/Part5/D.1/). It was in examples in opcua-python. In specification i have found out that it is also possible to define custom data types with help DataTypeDefinition(https://reference.opcfoundation.org/v104/Core/docs/Part3/8.48/) since release 1.04. 
I want to ask if this method of custom type deffinition is realised in opcu-python?

Thanks in advance!

Andrii

**Comment by oroulet** [comment, marked as answer]
it is implemented in opcua-asyncio. you can use the sync API and it should work . Much easier to work with new definitions

**Reply by swamper123** [reply]
Link to repo: [opcua-asyncio](https://github.com/FreeOpcUa/opcua-asyncio)

**Comment by AndreyKl4** [comment]
@oroulet @swamper123 Thanks a lot for your answers! Can you send me please an example or link in what way i have to define custom data types with help DataTypeDefinition.

**Reply by AndreasHeine** [reply]
https://github.com/FreeOpcUa/opcua-asyncio/tree/master/examples

**Reply by AndreyKl4** [reply]
@AndreasHeine Thank you!!!
